theia/templates/statefulset.yaml

```diff
@@ -0,0 +1,98 @@
+apiVersion: apps/v1
+kind: StatefulSet
```

Comment:
I would switch to a deployment, statefulsets usually give more incompatible changes.
If statefulset is the theia recommendation, I would create the PVC individually, and attach it as a volume. If not, a change on the volume size for example, would require full recreation.

Reply:
I tried that first, but had a problem during upgrades, the new pod would wait forever due to the volume being attached to the older pod. Is there anything similar to the recreate strategy for deployments?

Reply:
Deployments also have the `Recreate` upgrade policy.
```
  strategy:
        type: Recreate
```